tests/Oryx.Integration.Tests/Php/PhpPreRunCommandOrScriptTest.cs

```diff
@@ -0,0 +1,95 @@
+
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Tests.Common;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Oryx.Integration.Tests
+{
+    [Trait("category", "php")]
+    public class PhpPreRunCommandOrScriptTest : PhpEndToEndTestsBase
+    {
+        public PhpPreRunCommandOrScriptTest(ITestOutputHelper output, TestTempDirTestFixture fixture)
+            : base(output, fixture)
+        {
+        }
+
+        [Fact]
+        public async Task TwigExampleCanBuildAndRun_UsingPreRunCommand()
+        {
+            // Arrange
+            var phpVersion = "7.4";
+            var appName = "twig-example";
+            var hostDir = Path.Combine(_hostSamplesDir, "php", appName);
+            var volume = DockerVolume.CreateMirror(hostDir);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = $"{appDir}/myoutputdir"; 
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand($"oryx build {appDir} --platform php --language-version {phpVersion} -o {appOutputDir}")
+               .ToString();
+            var runScript = new ShellScriptBuilder()
+                .SetEnvironmentVariable(FilePaths.PreRunCommandEnvVarName, $"touch \"{appOutputDir}/test_pre_run.txt\"")
+                .AddCommand($"oryx create-script -appPath {appOutputDir} -output {RunScriptPath}")
+                .AddFileExistsCheck($"{appOutputDir}/test_pre_run.txt")
+                .AddCommand(RunScriptPath)
+                .ToString();
+
+            // Act & Assert
+            await EndToEndTestHelper.BuildRunAndAssertAppAsync(
+                appName, _output, volume,
+                "/bin/sh", new[] { "-c", buildScript },
+                _imageHelper.GetTestRuntimeImage("php", phpVersion),
+                ContainerPort,
+                "/bin/sh", new[] { "-c", runScript },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
+                    Assert.Contains("<h1>Hello World!</h1>", data);
+                });
+        }
+
+        [Fact]
+        public async Task TwigExampleCanBuildAndRun_UsingPreRunScript()
+        {
+            // Arrange
+            var phpVersion = "7.4";
+            var appName = "twig-example";
+            var hostDir = Path.Combine(_hostSamplesDir, "php", appName);
+            var volume = DockerVolume.CreateMirror(hostDir);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = $"{appDir}/myoutputdir"; 
+            var buildScript = new ShellScriptBuilder()
+               .AddCommand($"oryx build {appDir} --platform php --language-version {phpVersion} -o {appOutputDir}")
+               .ToString();
+            var runScript = new ShellScriptBuilder()
+                .SetEnvironmentVariable(FilePaths.PreRunCommandEnvVarName, "./prerunscript.sh")
```

Comment:
This is fine for now, but as @gpcastro was mentioning, probably installing a PHP extension that we do not include as part of the image would be good scenario to test.